### PR TITLE
Insurance deactivate

### PR DESCRIPTION
**Pull request by juggernautsei**
I had to do the same thing that Terry is working on but I did it a little different to make it so that insurances can be deactivated and reactivated.

### CODE REVIEW

**Comment by juggernautsei** [comment]
I have rebased this commit for inclusion in the 5.0

File: sql/database.sql [hunk 1]
```diff
@@ -2502,6 +2502,7 @@ CREATE TABLE `insurance_companies` (
   `x12_receiver_id` varchar(25) default NULL,
   `x12_default_partner_id` int(11) default NULL,
   `alt_cms_id` varchar(15) NOT NULL DEFAULT '',
+  `inactive` int(1) NOT NULL,
```
Comment (bradymiller) [inline sql/database.sql]: default this setting to 0
``````
`inactive` int(1) NOT NULL DEFAULT 0,

File: templates/insurance_companies/general_list.html [hunk 2]
```diff
@@ -6,12 +6,14 @@
...
 		<th width="300px"><b>{xl t='City, State'}</b></th>
 		<th><b>{xl t='Default X12 Partner'}</b></th>
+		<th><b>{xl t='Deactivated'}</b></th>
 	</tr>
 	{foreach from=$icompanies item=insurancecompany}
 	<tr height="22">
 		<td><a href="{$CURRENT_ACTION}action=edit&id={$insurancecompany->id}" onsubmit="return top.restoreSession()">{$insurancecompany->name}&nbsp;</a></td>
 		<td>{$insurancecompany->address->city} {$insurancecompany->address->state|upper}&nbsp;</td>
 		<td>{$insurancecompany->get_x12_default_partner_name()}&nbsp;</td>
+		<td>{if $insurancecompany->get_inactive() eq 1}Yes{/if}&nbsp;</td>
```
Comment (bradymiller) [inline templates/insurance_companies/general_list.html]: Translate Yes:
``````
<td>{if $insurancecompany->get_inactive() eq 1}{xl t='Yes'}{/if}&nbsp;</td>

File: sql/database.sql [hunk 3]
```diff
@@ -2502,6 +2502,7 @@ CREATE TABLE `insurance_companies` (
...
   PRIMARY KEY  (`id`)
 ) ENGINE=InnoDB;
```
Comment (bradymiller) [inline sql/database.sql]: Also need to add this entry in the sql upgrade script at the bottom of 4_2_2-to-5_0_0_upgrade.sql:
``````
#IfMissingColumn insurance_companies inactive
	ALTER TABLE `insurance_companies` ADD `inactive` int(1) NOT NULL DEFAULT 0;
#EndIf

File: library/patient.inc [hunk 4]
```diff
@@ -77,7 +77,7 @@ function getInsuranceProviders() {
     $returnval = array();
 
     if (true) {
-        $sql = "select name, id from insurance_companies order by name, id";
+        $sql = "select name, id from insurance_companies where inactive != 1 order by name, id";
         $rez = sqlStatement($sql);
         for($iter=0; $row=sqlFetchArray($rez); $iter++) {
```
Comment (bradymiller) [inline library/patient.inc]: The getInsuranceProvidersExtra() function below this also needs to be modified to only return the active insurance companies.

**Comment by bradymiller** [comment]
Review is complete. If you address the above issues, then it should be ready to go into the main codebase. thanks, -brady

**Comment by juggernautsei** [comment]
The issues above have been addressed and ready at your leisure

**Comment by bradymiller** [comment]
@juggernautsei ,
Looks like you still need to push your commit with your changes :)
-brady

**Comment by juggernautsei** [comment]
I don't know what to tell you when I drill down from within git hub I see the changes.
https://github.com/juggernautsei/openemr/blob/insurance_deactivate/library/patient.inc
Line 117 is added.

**Comment by bradymiller** [comment]
You brought that into another branch:
insurance_deactivate:
https://github.com/juggernautsei/openemr/commits/insurance_deactivate

Note that this PR is instead for the following branch:
Insurance_deactivate:
https://github.com/juggernautsei/openemr/commits/Insurance_deactivate

They are basically the same branches, except that Insurance_deactivate is missing the recent commit. Can fix with following on commandline:
git checkout Insurance_deactivate
git merge insurance_deactivate
git push origin

**Comment by juggernautsei** [comment]
sherwingaddis@WIN-890JAAMGPN0 ~/openemr/openemr (Insurance_deactivate)
$ git push origin
fatal: The current branch Insurance_deactivate has no upstream branch.
To push the current branch and set the remote as upstream, use

    git push --set-upstream origin Insurance_deactivate

**Comment by bradymiller** [comment]
ok, then this should do it:
git checkout Insurance_deactivate
git push origin Insurance_deactivate

**Comment by bradymiller** [comment]
Hi @juggernautsei,
I just brought your feature(including your most recent commit) into the main codebase. Thanks for the nice feature!
https://github.com/openemr/openemr/commit/be1216257af016997740fabd91be521702984d1a
-brady

**Pull request closed without merging**